### PR TITLE
[4.0] [com_csp] convert to prepared statement

### PR DESCRIPTION
**Pull request by alikon**
### Summary of Changes
use prepared statement for SQL

### Testing Instructions
test com_csp

### Expected result
should work as before

### Actual result
N/A

### CODE REVIEW

File: components/com_csp/Controller/ReportController.php [hunk 1]
```diff
@@ -108,10 +108,13 @@ private function isEntryExisting($report)
 
 		$query
 			->select('count(*)')
```
Comment (Quy) [inline components/com_csp/Controller/ReportController.php]:
```suggestion
			->select('COUNT(*)')
```

File: administrator/components/com_csp/Model/ReportsModel.php [hunk 2]
```diff
@@ -102,18 +103,20 @@ protected function getListQuery()
 		// Select the required fields from the table.
 		$query
 			->select('*')
-			->from('#__csp AS a');
+			->from($db->quoteName('#__csp'), 'a');
```
Comment (Quy) [inline administrator/components/com_csp/Model/ReportsModel.php]:
```suggestion
			->from($db->quoteName('#__csp', 'a'));
```

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on ce71a734c4a818b00fa7d4c8cf1905d0119a1ab6<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25188">issues.joomla.org/tracker/joomla-cms/25188</a>.</sub>

**Review by zero-24 — APPROVED** [review]

**Comment by SharkyKZ** [comment]
I have tested this item :white_check_mark: successfully on f00aa42d241d5ebb14a71dad760239f9fb47b392<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25188">issues.joomla.org/tracker/joomla-cms/25188</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25188">issues.joomla.org/tracker/joomla-cms/25188</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**